developing/languages/GO.md

```diff
@@ -4,9 +4,184 @@
 
 Go aka Golang is a programming language designed by folks at Google.
 
-## Resources
+This guide provides some general resources for working in Go.
+[Web Application in Go](../../web/server/go.md)
+provides some specifics to web development.
+
+## Learning Resources
+
+### References
 
 * [The Go Programming Language Official Website](https://golang.org/)
+* [Effective Go](https://golang.org/doc/effective_go.html) (how to do things “the Go way”)
 * [Go Pointer Primer](https://github.com/trussworks/go-pointer-primer)
+* [GoDoc](https://godoc.org/) (where you can read the docs for nearly any Go package)
+* [Go wiki](https://github.com/golang/go/wiki/Learn)
+* _Book_: [The Go Programming Language](http://www.gopl.io/)
+* Advanced Testing with Go
+  [Video](https://www.youtube.com/watch?v=yszygk1cpEc)
+  and [Article](https://about.sourcegraph.com/go/advanced-testing-in-go) (great overview of useful techniques, useful for all Go programmers)
+
+### Tours/Lessons
+
+* [A Tour of Go](https://tour.golang.org) (in-browser interactive language tutorial)
+* [How to Write Go Code](https://golang.org/doc/code.html) (info about the Go environment, testing, etc.)
 * [Go by Example](https://gobyexample.com)
 * [Go Track on Exercism](https://exercism.io/tracks/go)
+
+## Testing
+
+### General
+
+* Use table-driven tests where appropriate.
+* Make judicious use of helper functions so that the intent of a test is not lost in a sea of error checking and boilerplate.
+  Use [`t.Helper()`](https://golang.org/pkg/testing/#T.Helper) in your test helper functions to keep stack traces clean.
+
+### Coverage
+
+* Always test exported functions.
+  Exported functions should be treated as an API layer for other packages.
+  Cover the expected behavior and error scenarios as a user of that API.
+* Try not to test unexported functions.
+  Unexported functions are implementation details of exported ones
+  and should not change the intended usage.
+  If you find that an unexported function is complex and needs testing,
+  it might mean it needs to be refactored as it's exported function elsewhere.
+
+## Packages
```

Comment:
This might be a good starting point: https://github.com/golang-standards/project-layout
On MilMove we have been discussing whether our physical architecture (e.g. package directory structure) should match features more closely. That is, the go equivalent of redux ducks. This would allow teams clearer ownership of features and modules in the code and make it easier to see when they are touching code that effects other teams.